functional_unit_testing/parteh/py_modules/SyntheticBoundaries.py

```diff
@@ -0,0 +1,175 @@
+    # This method was provided by Charlie Koven via is inferences from the PPA
+    # literature.  Here, net daily carbon [kg] is based on one of two excluding
+    # parmaters (NPP per crown area per year), for plants that are either in
+    # the upper canopy (access to sunlight) or in the understory (low sunlight)
+    #
+    # c_area, footprint of the crown area [m2].
+    # presc_npp_canopy, npp generated per crown area in canopy         [kgC/m2/yr]
+    # presc_npp_understory, npp generated per crown area in understory [kgC/m2/yr]
+    # presc_nflux_p1, Nitrogen flux per crown area                     [kgN/m2/yr]
+    # presc_pflux_p1, Phosphorous flux per crown area                  [kgP/m2/yr]
+    # -----------------------------------------------------------------------------------
+
+    if( (phen_type == 1) or (leaf_status ==2)):
+        NetDailyC = presc_npp_p1 * c_area / day_per_year        
+        NetDailyN = presc_nflux_p1 * c_area / day_per_year
+        NetDailyP = presc_pflux_p1 * c_area / day_per_year
+    else:
+        NetDailyC = 0.0
+        NetDailyN = 0.0
+        NetDailyP = 0.0
+
+    return NetDailyC, NetDailyN, NetDailyP
+
+
+def DailyCNPFromStorageSinWave(doy,store_c,presc_npp_p1, \
+                               presc_nflux_p1,presc_pflux_p1,c_area,presc_npp_amp, \
+                               phen_type, leaf_status):
+
+
+    # This method is supposed to simulate a seasonal cycle of NPP
+    # In some cases we pass negative daily carbon gain to the allocation model
+    # however, we have to be careful to not make negative gains larger
+    # than available storage in those cases.  This is not necessarily the most
+    # realistic model, but its important to test that the parteh algorithms can handle
+    # these stressfull negative gain conditions.
+
+    doy0=0.0
+
+    sin_func = np.sin( (doy-doy0)/366.0 * 2.0 * np.pi )
+    
+    #if (sin_func>0.0):
+    #    NetDailyC = sin_func * presc_npp_p1 * c_area / day_per_year
+    #else:
+    #    NetDailyC = -np.minimum( -neg_store_frac * sin_func * presc_npp_p1* c_area / day_per_year, 0.98* np.float(store_c))
+
+    NetDailyC = (presc_npp_amp * sin_func * presc_npp_p1 + presc_npp_p1) * c_area/day_per_year
+
+    # This is a fail-safe, for large negatives, cant be larger than storage
+    
+    if (NetDailyC < 0.0):
+        NetDailyC = -np.minimum(-NetDailyC,0.98* np.float(store_c))
+
+    #print("sin_func: {}, NetDailyC: {}, store_c: {}, c_area :{}".format(sin_func,NetDailyC,store_c,c_area))
+
+    if( (phen_type == 1) or (leaf_status ==2)):
+        NetDailyN = presc_nflux_p1 * c_area / day_per_year
+        NetDailyP = presc_pflux_p1 * c_area / day_per_year
+    else:
+        NetDailyN = 0.0
+        NetDailyP = 0.0
+        NetDailyC = 0.0
+    
+    return NetDailyC, NetDailyN, NetDailyP
+
+
+def DeciduousPhenology(doy, target_leaf_c, store_c, phen_type):
+
+    # Time leaf-on with rising NPP
+    leaf_on_doy  = np.int(366.0 * 0.01)
```

Comment:
Is this for all deciduous PFTs or just for use with the prescribed phenology version? I was just trying to figure out how PARTEH handles cold dec phenology for the sake of coordinating with CLM...